notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java

```diff
@@ -943,15 +948,11 @@ public void removeDraft(String draftName) throws WikiException {
    */
   @Override
   public void removeDraftById(String draftId) throws Exception {
-    Page draftNote = dataStorage.getDraftPageById(draftId);
+    DraftPage draftNote = dataStorage.getDraftPageById(draftId);
+    deleteNoteMetadataProperties(draftNote, draftNote.getLang(), NOTE_METADATA_DRAFT_PAGE_OBJECT_TYPE);
```

Comment:
I prefer that the properties deletion come after the draft deletion itself, so when there is a problem with the draft deletion the properties will not be deleted

Reply:
Nope, we need to delete properties before, because when we delete them we need to fetch the draft an compare illustrations use, so we don't want to fetch a deleted draft